src/Common/ServerUUIDFile.cpp

```diff
@@ -0,0 +1,126 @@
+#include "ServerUUIDFile.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <sys/file.h>
+
+#include <Poco/File.h>
+#include <Poco/UUID.h>
+#include <common/errnoToString.h>
+#include <common/logger_useful.h>
+
+#include <IO/LimitReadBuffer.h>
+#include <IO/Operators.h>
+#include <IO/ReadBufferFromFile.h>
+#include <IO/WriteBufferFromFileDescriptor.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_OPEN_FILE;
+    extern const int CANNOT_CLOSE_FILE;
+    extern const int CANNOT_TRUNCATE_FILE;
+    extern const int CANNOT_SEEK_THROUGH_FILE;
+}
+
+
+ServerUUIDFile::FillFunction ServerUUIDFile::write_server_uuid = [](WriteBuffer & out)
+{
+    union
+    {
+        char bytes[16];
+        struct
+        {
+            UInt64 a;
+            UInt64 b;
+        } words;
+        __uint128_t uuid;
+    } random;
+
+    random.words.a = thread_local_rng(); //-V656
+    random.words.b = thread_local_rng(); //-V656
+
+    struct ServerUUID : Poco::UUID
+    {
+        ServerUUID(const char * bytes, Poco::UUID::Version version) : Poco::UUID(bytes, version) { }
+    };
+
+    auto server_uuid = ServerUUID(random.bytes, Poco::UUID::UUID_RANDOM).toString();
+
+    out << server_uuid;
+};
+
+
+ServerUUIDFile::ServerUUIDFile(std::string path_, FillFunction fill_) : path(std::move(path_)), fill(std::move(fill_))
+{
+    /// If file already exists. NOTE Minor race condition.
+    if (Poco::File(path).exists())
+    {
+        std::string contents;
+        {
+            ReadBufferFromFile in(path, 1024);
+            LimitReadBuffer limit_in(in, 1024, false);
+            readStringUntilEOF(contents, limit_in);
+        }
+
+        if (!contents.empty())
+            LOG_INFO(
+                &Poco::Logger::get("ServerUUIDFile"),
+                "Server UUID file {} already exists - unclean restart. Contents:\n{}",
```

Comment:
No... we should generate uuid file if it does not exist and use the existing file if it does.